src/source/JavaGenerator.scala

```diff
@@ -249,6 +249,35 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
     val javaName = if (r.ext.java) (ident.name + "_base") else ident.name
     val javaFinal = if (!r.ext.java && spec.javaUseFinalForRecord) "final " else ""
 
+    def checkIfRecordContainsSets(r: Record): Boolean = {
```

Comment:
nit: I'd prefer that predicates like this are named as a question or statement.  E.g. recordContainsSets() or doesRecordContainSets().